app/javascript/test/SpellingSuggestion.test.tsx

```diff
@@ -1,21 +1,254 @@
 
 jest.mock('i18n-js', () => {
   return jest.requireActual('i18n-js/dist/require/index');
 });
 
-describe('Alert component', () => {
+describe('SpellingSuggestion component', () => {
+  const headers = {
+    Accept: 'application/json',
+    'Content-Type': 'application/json'
+  };
+  global.fetch = jest.fn(() =>
+    Promise.resolve({
+      json: () => Promise.resolve({})
+    })
+  ) as jest.Mock;
+
   const spellingSuggestionProps = {
-    suggested: '<a href="/search?affiliate=test_alert&amp;query=government">government</a>',
-    original: '<a href="/search?affiliate=test_alert&amp;query=%2Bgovermment">govermment</a>'
+    suggested: '<a class="suggestedQuery" href="/search?affiliate=test_affiliate&amp;query=medical">medical</a>',
+    original: '<a class="originalQuery" href="/search?affiliate=test_affiliate&amp;query=mecidal">mecidal</a>',
+    originalUrl: '/search?affiliate=test_affiliate&query=mecidal',
+    originalQuery: 'mecidal',
+    suggestedQuery: 'medical',
+    suggestedUrl: '/search?affiliate=test_affiliate&query=medical',
+    affiliate: 'test_affiliate'
   };
 
   it('renders SpellingSuggestion component', () => {
     render(
-      <SpellingSuggestion {...spellingSuggestionProps} />
+      <LanguageContext.Provider value={i18n} >
+        <SpellingSuggestion {...spellingSuggestionProps} vertical='web' />
+      </LanguageContext.Provider>
+    );
+  });
+
+  it('Show original and suggested query', () => {
+    render(
+      <LanguageContext.Provider value={i18n} >
+        <SpellingSuggestion {...spellingSuggestionProps} vertical='web'/>
+      </LanguageContext.Provider>
+    );
+
+    expect(screen.getByText('Showing results for')).toBeInTheDocument();
+    expect(screen.getByText('medical')).toBeInTheDocument();
+    const [suggestedQuery] = Array.from(document.getElementsByClassName('suggestedQuery'));
+    expect(suggestedQuery).toHaveAttribute('href', '/search?affiliate=test_affiliate&query=medical');
+
+    expect(screen.getByText('Search instead for')).toBeInTheDocument();
+    expect(screen.getByText('mecidal')).toBeInTheDocument();
+    const [originalQuery] = Array.from(document.getElementsByClassName('originalQuery'));
+    expect(originalQuery).toHaveAttribute('href', '/search?affiliate=test_affiliate&query=mecidal');
+  });
+
+  it('clickTracking for suggestedQuery and originalQuery: web vertical', () => {
+    render(
+      <LanguageContext.Provider value={i18n} >
+        <SpellingSuggestion {...spellingSuggestionProps} vertical='web'/>
+      </LanguageContext.Provider>
+    );
+
+    const suggestedQueryLink = screen.getByText(/medical/i);
+    fireEvent.click(suggestedQueryLink);
+    const clickBodySuggestedQuery = {
+      affiliate: 'test_affiliate',
+      url: 'http://localhost/search?affiliate=test_affiliate&query=medical',
+      module_code: 'BSPEL',
+      position: 1,
+      query: 'medical',
+      vertical: 'web'
+    };
+ 
+    expect(fetch).toHaveBeenCalledWith('/clicked', {
```

Comment:
I see this happening very repeatedly. What do you think about creating a helper something like

```js
function expectFetchtoHaveBeenCalledWith(data) {
   expect(fetch).toHaveBeenCalledWith('/clicked', {
      body: JSON.stringify(data),
      headers,
      method: 'POST',
      mode: 'cors'
    });
}

...
expectFetchtoHaveBeenCalledWith(clickBodySuggestedQuery)
```

Reply:
So we are using this across many tests, so I think this will need to refactored across all the tests where click tracking is being tested. thoughts?